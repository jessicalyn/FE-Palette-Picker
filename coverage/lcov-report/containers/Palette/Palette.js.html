<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for containers/Palette/Palette.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">containers/Palette</a> Palette.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">63.41% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>26/41</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">68.75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>11/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">61.54% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>24/39</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a>
<a name='L90'></a><a href='#L90'>90</a>
<a name='L91'></a><a href='#L91'>91</a>
<a name='L92'></a><a href='#L92'>92</a>
<a name='L93'></a><a href='#L93'>93</a>
<a name='L94'></a><a href='#L94'>94</a>
<a name='L95'></a><a href='#L95'>95</a>
<a name='L96'></a><a href='#L96'>96</a>
<a name='L97'></a><a href='#L97'>97</a>
<a name='L98'></a><a href='#L98'>98</a>
<a name='L99'></a><a href='#L99'>99</a>
<a name='L100'></a><a href='#L100'>100</a>
<a name='L101'></a><a href='#L101'>101</a>
<a name='L102'></a><a href='#L102'>102</a>
<a name='L103'></a><a href='#L103'>103</a>
<a name='L104'></a><a href='#L104'>104</a>
<a name='L105'></a><a href='#L105'>105</a>
<a name='L106'></a><a href='#L106'>106</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">25x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { connect } from 'react-redux';
import Color from '../Color/Color';
import { storeColors } from '../../actions';
import generate from '../../assets/generate.png';
import { fetchOptionsCreator } from '../../utils/fetchOptionsCreator';
import { postNewPalette } from '../../thunks/postNewPalette';
&nbsp;
export class Palette extends Component {
  constructor() {
    super();
    this.state = {
      palette_name: '',
      project_id: 0,
      color_1: '',
      color_2: '',
      color_3: '',
      color_4: '',
      color_5: ''
    }
  }
  
  updateColors = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
    const updatedColors = <span class="cstat-no" title="statement not covered" >this.props.currentColors.map(<span class="fstat-no" title="function not covered" >co</span>lor =&gt; {</span>
<span class="cstat-no" title="statement not covered" >      if (color.locked) {</span>
<span class="cstat-no" title="statement not covered" >        return color</span>
      } else {
        const newColor = <span class="cstat-no" title="statement not covered" >Math.floor(Math.random()*16777215).toString(16)</span>
<span class="cstat-no" title="statement not covered" >        return { color: `#${newColor}`, locked: false }</span>
      }
    })
<span class="cstat-no" title="statement not covered" >    this.props.storeColors(updatedColors)</span>
  }
&nbsp;
  handleToggle = (e) =&gt; {
    const colorId = parseInt(e.target.id)
    const updatedColors = this.props.currentColors.map((color, index) =&gt; {
      <span class="missing-if-branch" title="if path not taken" >I</span>if (index === colorId) {
<span class="cstat-no" title="statement not covered" >        return { color: color.color , locked: !color.locked }</span>
      } else {
        return color
      }
    })
    this.props.storeColors(updatedColors)
  }
&nbsp;
  setPalette = <span class="fstat-no" title="function not covered" >(e</span>) =&gt; {
<span class="cstat-no" title="statement not covered" >    e.preventDefault()</span>
<span class="cstat-no" title="statement not covered" >    this.props.currentColors.forEach(<span class="fstat-no" title="function not covered" >(c</span>olor, index) =&gt; {</span>
      let colorKey = <span class="cstat-no" title="statement not covered" >`color_${index + 1}`</span>
<span class="cstat-no" title="statement not covered" >      this.setState({ [colorKey]: color.color })</span>
    })
<span class="cstat-no" title="statement not covered" >    this.addNewPalette()</span>
  }
&nbsp;
  addNewPalette = <span class="fstat-no" title="function not covered" >as</span>ync () =&gt; {
    const body = <span class="cstat-no" title="statement not covered" >this.state</span>
    const options = <span class="cstat-no" title="statement not covered" >await fetchOptionsCreator('POST', body)</span>
<span class="cstat-no" title="statement not covered" >    this.props.postNewPalette(options, body)</span>
  }
&nbsp;
  handleChange = (e) =&gt; {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }
&nbsp;
  render() {
    const colorSwatches = this.props.currentColors.map((color, index) =&gt; {
      return &lt;Color key={color.color} {...color} index={index} handleToggle={this.handleToggle} /&gt;
    })
    const projectList = this.props.projects.map((project, index) =&gt; {
      let projectName = project.project_name
      return &lt;option key={`option-${projectName}-${index}`} value={project.project_id} &gt;{projectName}&lt;/option&gt;
    })
    return(
      &lt;div className="colors-wrapper"&gt;
        &lt;h2&gt;Generate New Palette
          &lt;button onClick={this.updateColors}&gt;
            &lt;img src={generate} alt={'Generate New Palette icon'} /&gt;
          &lt;/button&gt;
        &lt;/h2&gt;
        &lt;select value={this.state.project_id} name="project_id" onChange={this.handleChange}&gt;
          &lt;option value="0" disable selected default&gt;Select a Project&lt;/option&gt;
          {projectList}
        &lt;/select&gt;
        &lt;form onSubmit={this.setPalette}&gt;
          &lt;input type="text" onChange={this.handleChange} value={this.state.palette_name} name="palette_name" /&gt;
          &lt;button className="palette-button" disabled={this.state.project_id == 0}&gt;Save Palette to Project&lt;/button&gt;
        &lt;/form&gt;
        {colorSwatches}
      &lt;/div&gt;
    )
  }
}
&nbsp;
export const mapStateToProps = (state) =&gt; ({
  currentColors: state.currentColors,
  projects: state.projects
})
&nbsp;
export const mapDispatchToProps = (dispatch) =&gt; ({
  storeColors: (colors) =&gt; dispatch(storeColors(colors)),
  postNewPalette: (options, body) =&gt; dispatch(postNewPalette(options, body)),
})
&nbsp;
export default connect(mapStateToProps, mapDispatchToProps)(Palette);</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed May 15 2019 15:06:53 GMT-0600 (Mountain Daylight Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
