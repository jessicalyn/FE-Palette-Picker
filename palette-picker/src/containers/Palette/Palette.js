import React, { Component } from 'react';
import { connect } from 'react-redux';
import Color from '../Color/Color';
import { storeColors } from '../../actions';
import generate from '../../assets/generate.png';
import { fetchOptionsCreator } from '../../utils/fetchOptionsCreator';
import { postNewPalette } from '../../thunks/postNewPalette';

export class Palette extends Component {
  constructor() {
    super();
    this.state = {
      palette_name: '',
      project_id: 0,
      color_1: '',
      color_2: '',
      color_3: '',
      color_4: '',
      color_5: ''
    }
  }
  
  updateColors = () => {
    const updatedColors = this.props.currentColors.map(color => {
      if (color.locked) {
        return color
      } else {
        const newColor = Math.floor(Math.random()*16777215).toString(16)
        return { color: `#${newColor}`, locked: false }
      }
    })
    this.props.storeColors(updatedColors)
  }

  handleToggle = (e) => {
    const colorId = parseInt(e.target.id)
    const updatedColors = this.props.currentColors.map((color, index) => {
      if (index === colorId) {
        return { color: color.color , locked: !color.locked }
      } else {
        return color
      }
    })
    this.props.storeColors(updatedColors)
  }

  setPalette = async (e) => {
    e.preventDefault()
    await this.props.currentColors.forEach((color, index) => {
      let colorKey = `color_${index + 1}`
      this.setState({ [colorKey]: color.color })
    })
    this.addNewPalette()
  }

  addNewPalette = async () => {
    const body = this.state
    const options = await fetchOptionsCreator('POST', body)
    await this.props.postNewPalette(options, body)
  }

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }

  render() {
    const colorSwatches = this.props.currentColors.map((color, index) => {
      return <Color key={color.color} {...color} index={index} handleToggle={this.handleToggle} />
    })
    const projectList = this.props.projects.map((project, index) => {
      let projectName = project.project_name
      return <option key={`option-${projectName}-${index}`} value={project.project_id} >{projectName}</option>
    })
    return(
      <div className="colors-wrapper">
        <h2>Generate New Palette
          <button onClick={this.updateColors}>
            <img src={generate} alt={'Generate New Palette icon'} />
          </button>
        </h2>
        <select value={this.state.project_id} name="project_id" onChange={this.handleChange}>
          <option value="0" disable selected default>Select a Project</option>
          {projectList}
        </select>
        <form onSubmit={this.setPalette}>
          <input type="text" onChange={this.handleChange} value={this.state.palette_name} name="palette_name" />
          <button className="palette-button" disabled={this.state.project_id == 0}>Save Palette to Project</button>
        </form>
        {colorSwatches}
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  currentColors: state.currentColors,
  projects: state.projects
})

export const mapDispatchToProps = (dispatch) => ({
  storeColors: (colors) => dispatch(storeColors(colors)),
  postNewPalette: (options, body) => dispatch(postNewPalette(options, body))
})

export default connect(mapStateToProps, mapDispatchToProps)(Palette);