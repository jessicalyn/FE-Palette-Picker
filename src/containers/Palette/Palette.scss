.generate-palette-text {
  display: flex;
  flex-direction: column;
  align-items: center;
  font-family: $font-main;
  margin-right: 15px;
}

.generate-button {
  width: 75px;
  height: 75px;
}

.colors-wrapper{
  display: flex;
  flex-direction: column;
}