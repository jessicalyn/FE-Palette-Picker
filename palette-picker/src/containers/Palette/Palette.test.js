import { Palette, mapStateToProps, mapDispatchToProps } from './Palette';
import React from 'react';
import { shallow } from 'enzyme';
import { fetchOptionsCreator } from '../../utils/fetchOptionsCreator';

jest.mock('../../actions')
jest.mock('../../thunks/postNewPalette')

describe('Palette Container', () => {
  describe('Palette', () => {
    let wrapper;
    const mockCurrentColors = [
      {"color": "#f82e41", "locked": true}, 
      {"color": "#2261cd", "locked": true}, 
      {"color": "#271de3", "locked": true}, 
      {"color": "#1563ec", "locked": true}, 
      {"color": "#9e0246", "locked": true}
    ]
    const mockProjects = [
      { project_id: 1,
        project_name: "Travel App",
        created_at: "2019-05-10T17:45:09.612Z",
        updated_at: "2019-05-10T17:45:09.612Z"
      },
      { project_id: 2,
        project_name: "Fitness App",
        created_at: "2019-05-10T17:45:09.634Z",
        updated_at: "2019-05-10T17:45:09.634Z"
      }
    ]
    let mockStoreColors
    let mockPostNewPalette
    const mockState = {
      palette_name: '',
      project_id: 0,
      color_1: '',
      color_2: '',
      color_3: '',
      color_4: '',
      color_5: ''
    }

    const mockPalette = {
      palette_name: 'Summer colors',
      project_id: 0,
      color_1: '#f82e41',
      color_2: '#2261cd',
      color_3: '#271de3',
      color_4: '#f82e41',
      color_5: '#271de3'
    }

    beforeEach(() => {
      mockStoreColors = jest.fn().mockImplementation(() => Promise.resolve({mockCurrentColors}))
      mockPostNewPalette = jest.fn().mockImplementation(() => Promise.resolve({mockPalette}))
      wrapper = shallow(
        <Palette
          currentColors={mockCurrentColors}
          projects={mockProjects}
          storeColors={mockStoreColors} 
          postNewPalette={mockPostNewPalette}
        />
      )
    })

    it('should match the snapshot', () => {
      expect(wrapper.debug()).toMatchSnapshot()
    })

    it('should have proper default state', () => {
      expect(wrapper.state()).toEqual(mockState)
    })

    it('should call props function storeColors with updatedColors array', () => {
      const mockEvent = { target: { id : 10 }}
      const mockUpdatedColors = [
        {"color": "#f82e41", "locked": true}, 
        {"color": "#2261cd", "locked": true}, 
        {"color": "#271de3", "locked": true}, 
        {"color": "#1563ec", "locked": true}, 
        {"color": "#9e0246", "locked": true}
      ]
      wrapper.instance().handleToggle(mockEvent)

      expect(wrapper.instance().props.storeColors).toHaveBeenCalledWith(mockUpdatedColors)
    })

    it.skip('should call props function postNewPalette with options and body', async () => {
      wrapper.setState({
        palette_name: 'Summer colors',
        project_id: 0,
        color_1: '#f82e41',
        color_2: '#2261cd',
        color_3: '#271de3',
        color_4: '#f82e41',
        color_5: '#271de3'
      })
      const body = wrapper.state()
      const options = await fetchOptionsCreator('POST', body)

      wrapper.instance().addNewPalette()
      expect(wrapper.instance().props.postNewPalette).toHaveBeenCalledWith(options, body)
    })

    it('should set state with a value', () => {
      const mockEvent = { target: { name : "palette_name", value: "MyNewPalette" } }
      wrapper.instance().handleChange(mockEvent)

      expect(wrapper.state('palette_name')).toEqual("MyNewPalette")
    })
  })

  describe('mapStateToProps', () => {
    it('should return an object with a currentColors array', () => {
      const mockState = {
        currentColors: [
          {"color": "#f82e41", "locked": true}, 
          {"color": "#2261cd", "locked": true}
        ],
        fakeState: "Not real state to return"
      }
      const expected = {
        currentColors: [
          {"color": "#f82e41", "locked": true}, 
          {"color": "#2261cd", "locked": true}
        ]
      }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should dispatch storeColors with an array of colors', () => {
      const mockColors = [
        {"color": "#f82e41", "locked": true}, 
        {"color": "#2261cd", "locked": true}
      ]
      const mockDispatch = jest.fn()
      const storeColors = jest.fn()
      const actionToDispatch = storeColors(mockColors)
      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.storeColors(mockColors)

      expect(mockDispatch).toHaveBeenCalled(actionToDispatch)
    })

    it('should dispatch postNewPalette with options and body', () => {
      const mockBody = {
        palette_name: 'Summer colors',
        project_id: 0,
        color_1: '#f82e41',
        color_2: '#2261cd',
        color_3: '#271de3',
        color_4: '#f82e41',
        color_5: '#271de3'
      }
      const options = fetchOptionsCreator('POST', mockBody)
      const mockDispatch = jest.fn()
      const postNewPalette  = jest.fn()
      const actionToDispatch = postNewPalette(options, mockBody)
      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.postNewPalette(options, mockBody)

      expect(mockDispatch).toHaveBeenCalled(actionToDispatch)
    })
  })
})