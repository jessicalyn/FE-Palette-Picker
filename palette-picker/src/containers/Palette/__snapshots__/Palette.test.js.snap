// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette Container Palette should match the snapshot 1`] = `
"<div className=\\"colors-wrapper\\">
  <h2>
    Generate New Palette
    <button onClick={[Function]}>
      <img src=\\"generate.png\\" alt=\\"Generate New Palette icon\\" />
    </button>
  </h2>
  <select value={0} name=\\"project_id\\" onChange={[Function]}>
    <option value=\\"0\\" disable={true} selected={true} default={true}>
      Select a Project
    </option>
    <option value={1}>
      Travel App
    </option>
    <option value={2}>
      Fitness App
    </option>
  </select>
  <form onSubmit={[Function]}>
    <input type=\\"text\\" onChange={[Function]} value=\\"\\" name=\\"palette_name\\" />
    <button className=\\"palette-button\\" disabled={true}>
      Save Palette to Project
    </button>
  </form>
  <Color color=\\"#f82e41\\" locked={true} index={0} handleToggle={[Function]} />
  <Color color=\\"#2261cd\\" locked={true} index={1} handleToggle={[Function]} />
  <Color color=\\"#271de3\\" locked={true} index={2} handleToggle={[Function]} />
  <Color color=\\"#1563ec\\" locked={true} index={3} handleToggle={[Function]} />
  <Color color=\\"#9e0246\\" locked={true} index={4} handleToggle={[Function]} />
</div>"
`;
